backend/src/main/java/ca/bc/gov/hlth/hnweb/security/KeycloakClientRoleConverter.java

```diff
@@ -1,44 +1,32 @@
 package ca.bc.gov.hlth.hnweb.security;
 
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.core.convert.converter.Converter;
 import org.springframework.security.core.GrantedAuthority;
 import org.springframework.security.core.authority.SimpleGrantedAuthority;
 import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.stereotype.Component;
 
-import java.util.Collection;
-import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
-
+@Component
 public class KeycloakClientRoleConverter implements Converter<Jwt, Collection<GrantedAuthority>> {
 
 	private static final String ROLE_PREFIX = "ROLE_";
-	// TODO (weskubo-cgi) Replace with the correct resource_access
-    private static final String API_CLIENT_NAME = "realm-management";
 
-    @SuppressWarnings("unchecked")
-	public Collection<GrantedAuthority> convert(final Jwt jwt) {
+    @Autowired
+    private SecurityProperties securityProperties;
 
-        Collection<GrantedAuthority> authorities = null;
-
-        /* The roles in the access token look like
-        "resource_access": {
-          "user-management-service": {
-            "roles": [
-               "view-groups"
-            ]
-          }
-        }*/
-        final Map<String, Object> resourceAccesses = (Map<String, Object>) jwt.getClaims().get("resource_access");
-        if (resourceAccesses != null) {
-            final Map<String, Object> resource = (Map<String, Object>) resourceAccesses.get(API_CLIENT_NAME);
-            if (resource != null) {
-                authorities = ((List<String>) resource.get("roles")).stream()
-                        .map(roleName -> ROLE_PREFIX + roleName) // prefix required to map to a Spring Security "role"
-                        .map(SimpleGrantedAuthority::new)
-                        .collect(Collectors.toList());
-            }
-        }
+	public Collection<GrantedAuthority> convert(final Jwt jwt) {
+    	List<String> permissions = SecurityUtil.loadPermissions(jwt, securityProperties.getRolePermissions());
+    	
+    	List<GrantedAuthority> authorities = permissions.stream()
```

Comment:
I think creating a variable here was just a carry over from the old method.
Could probably just return directly without creating a variable.

Reply:
Removed unnecessary local variable.